src/app/util/attribute-storage.cpp

```diff
@@ -523,6 +523,12 @@ EmberAfStatus emAfReadOrWriteAttribute(EmberAfAttributeSearchRecord * attRecord,
                                 uint8_t * attributeLocation =
                                     (am->mask & ATTRIBUTE_MASK_SINGLETON ? singletonAttributeLocation(am)
                                                                          : attributeData + attributeOffsetIndex);
+
+                                if (attributeOffsetIndex + readLength > sizeof(attributeData))
+                                {
+                                    return EMBER_ZCL_STATUS_INSUFFICIENT_SPACE;
+                                }
```

Comment:
This check doesn't seem right: `readLength` is the size available in the output buffer; there's no reason it should be no larger than the size of the attribute we are trying to read.  In particular, if the caller does not know how big the attribute it's trying to read is, having a conservatively-large size here is fine.